### PR TITLE
feat(Component=ButtonDropdown): Adding a button with a dropdown arrow

### PR DESCRIPTION
**Pull request by ryanballa**
Creating a button with a dropdown arrow for use in menus or other situations that need to show

expansion.

### CODE REVIEW

**Comment by ryanballa** [comment]
Screenshot 
screen shot 2016-12-12 at 5 01 00 pm

File: src/components/ButtonDropdown.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          verticalAlign: 'middle',
+          height: 0,
+          width: 0,
+        },
+      },
+    });
+    const content = this.props.children || this.props.text;
+
+    return (
+      <span>
```
Comment (KevinMarren) [inline src/components/ButtonDropdown.jsx]: why don't you use a button tag here? Seems much more semantic to do:
<button style={style.DropdownButton}>{content}<span style={style.Caret}></span</button>

File: specs/components/ButtonDropdown.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  it('Group Position: Center', () => this.props({ groupPosition: 'center' }));
+  it('Group Position: Right', () => this.props({ groupPosition: 'right' }));
+  it('Sets Styles', () => this.props({ style: { background: 'pink' } }));
+  it('Resets Styles', () => this.props({ style: {} }));
+
+  /**
+   * Documentation (Markdown)
+   */
+  this.footer(`
+  ### Button
```
Comment (KevinMarren) [inline specs/components/ButtonDropdown.spec.jsx]: ### Dropdown Button?

File: specs/components/ButtonDropdown.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+/* eslint func-names: "off" */
+
+import React from 'react';
+import loremIpsum from 'lorem-ipsum';
+import ButtonDropdown from '../../src/components/ButtonDropdown';
+
+describe('ButtonDropdown', function () {
```
Comment (KevinMarren) [inline specs/components/ButtonDropdown.spec.jsx]: For some reason DropdownButton reads better to me, I know that is an annoying change though

File: src/components/ButtonDropdown.jsx (hunk 1 above)
Comment (ryanballa) [inline src/components/ButtonDropdown.jsx]: I'm overriding the renderChildren method of Button which does get rendered into a <button> element. This [can be seen here](https://github.com/signal/sprinkles-ui/blob/master/src/components/Button.jsx#L167).

File: specs/components/ButtonDropdown.spec.jsx (hunk 3 above)
Comment (ryanballa) [inline specs/components/ButtonDropdown.spec.jsx]: I just wanted them easy to find by starting them all with the name 'Button'. There's going to be another. Not my favorite name either, perhaps I should group them in a folder.

**Comment by milemilo** [comment]
Possibly add prop for arrow rotation

**Pull request closed without merging**